src/mlpack/core/data/image_resize_crop.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file core/data/image_resize_crop.hpp
+ * @author Omar Shrit
+ *
+ * Image resize and crop functionalities.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+#define MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+
+
+/**
+ * Image resize/crop interfaces.
+ */
+
+/**
+ * Resize the images matrix.
+ *
+ * @param images The input matrix that contains the images to be resized.
+ * @param info Contains relevant input images information.
+ * @param resizedImages The output matrix the contains the resized images.
+ * @param newWidth The new requested width for the resized images.
+ * @param newHeight The new requested height for the resized images.
+ */
+template<typename eT>
+void ResizeImages(arma::Mat<eT>& images, data::ImageInfo& info,
+    arma::Mat resizedImages, const size_t newWidth, const size_t newHeight)
+{
+  if (images.n_rows != resizedImages and images.n_cols != resizedImages.n_cols)
+  {
+    std::ostringstream oss;
+    oss << "ResizeImage(): the resizedImage matrix need to have identical"
+      " dimensions to the images matrix" << std::endl;
+  }
+
+  for (size_t i = 0; i < images.n_cols; ++i)
+  {
+    stbir_resize_uint8(image.colptr(i), info.Width(), info.Height(), 0,
+                       resizedImages.colptr(i), newWidth, newHeight, 0,
+                       info.Channels());
+  }
```

Comment:
I think you would need to restructure this---if `eT != uint8_t`, this won't work very well:

```suggestion
  if (std::is_same_v<eT, uint8_t>)
  {
    for (size_t i = 0; i < images.n_cols; ++i)
    {
      stbir_resize_uint8(image.colptr(i), info.Width(), info.Height(), 0,
                         resizedImages.colptr(i), newWidth, newHeight, 0,
                         info.Channels());
    }
  }
  else if (std::is_same_v<eT, float>)
  {
    // use float version
  }
  else
  {
    arma::Mat<uint8_t> convImages = conv_to<arma::Mat<uint8_t>>::from(images);
    // call uint8 version
    // convert back to original `images`
  }
```

Anyway I think this is still a prototype, so, sorry if you are already thinking this and just haven't gotten there yet :smile:

Reply:
Yes, and I also I am afraid of handling the float version, I have no idea how the RGB is going to be there, and how the user is going to handle it. I need to think about it